rust/arrow/src/array/transform/mod.rs

```diff
@@ -687,4 +690,24 @@ mod tests {
         let expected = Int16Array::from(vec![Some(1), None]);
         assert_eq!(result.keys(), &expected);
     }
+
+    #[test]
+    fn test_binary_fixed_sized_offsets() {
+        let array =
+            FixedSizeBinaryArray::from(vec![vec![0, 0], vec![0, 1], vec![0, 2]]).data();
+        let array = array.slice(1, 2);
+
+        let arrays = vec![&array];
+
+        let mut mutable = MutableArrayData::new(arrays, false, 0);
+
+        mutable.extend(0, 1, 2);
+        mutable.extend(0, 0, 1);
+
+        let result = mutable.freeze();
+        let result = FixedSizeBinaryArray::from(Arc::new(result));
+
+        let expected = FixedSizeBinaryArray::from(vec![vec![0, 2], vec![0, 1]]);
```

Comment:
I am having trouble figuring out why this is the expected output. I expected something like

```
        let expected = FixedSizeBinaryArray::from(vec![vec![0, 1], vec![0, 0]]);
```

As I read 

```
        mutable.extend(0, 1, 2); // --> pick element 1 (aka vec[0, 1])
        mutable.extend(0, 0, 1); // --> pick element 0 (aka vec[0, 0])
```

Reply:
I am sorry, I should have documented the test better. I added a comment to it. Does it became understandable with the comment? The idea is to check that this works with arrays with offsets.

Reply:
oh, man, I missed the call to `        let array = array.slice(1, 2);` 👍  thanks @jorgecarleitao